Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM node:16-alpine
+ARG PNPM_VERSION=7.1.6
+RUN npm --location=global install pnpm@${PNPM_VERSION}
+
+WORKDIR /root/app
+COPY . .
+RUN pnpm i --prod
```

Comment:
to prevent dev dependencies to be installed. there are a lot of things we don't need such as cypress that gets installed if we don't pass that flag

Reply:
But if you need to build the app, you need the dev deps. Have you tested it?

Reply:
OK. I'd trust you then

Reply:
```shell
STEP 1/7: FROM node:16-alpine
Resolving %!q(<nil>) to docker.io (enforced by caller)
Trying to pull docker.io/library/node:16-alpine...
Getting image source signatures
Copying blob sha256:db20f7380b7cdc8c4a43c45537e1406491eec8682ebf1cfa8dbf054743e45026
Copying blob sha256:b3c136eddcbf2003d3180787cef00f39d46b9fd9e4623178282ad6a8d63ad3b0
Copying blob sha256:d4be7a92b9d4ce9556ebed99fd3a3ded1553a503782d4b4099a6dd3506fc35c7
Copying blob sha256:49fd8c1406ff3d95cf1a7c8c3040042630b2940cdc28c10a27bdfb52181f9272
Copying blob sha256:db20f7380b7cdc8c4a43c45537e1406491eec8682ebf1cfa8dbf054743e45026
Copying blob sha256:b3c136eddcbf2003d3180787cef00f39d46b9fd9e4623178282ad6a8d63ad3b0
Copying blob sha256:d4be7a92b9d4ce9556ebed99fd3a3ded1553a503782d4b4099a6dd3506fc35c7
Copying blob sha256:49fd8c1406ff3d95cf1a7c8c3040042630b2940cdc28c10a27bdfb52181f9272
Copying config sha256:e548f8c9983fd83ad9c4518a1c728440f5c1617d6514cef78dc1f7c19a7c607f
Writing manifest to image destination
Storing signatures
STEP 2/7: RUN corepack enable
--> 2303d0f9f78
STEP 3/7: WORKDIR /root/app
--> 58184d89fd9
STEP 4/7: COPY . .
--> 58d17f9dc13
STEP 5/7: RUN pnpm i --prod
Recreating /root/app/node_modules

   ╭─────────────────────────────────────────────────────────────────╮
   │                                                                 │
   │                Update available! 7.4.0 → 7.4.1.                 │
   │   Changelog: https://github.com/pnpm/pnpm/releases/tag/v7.4.1   │
   │     Run "corepack prepare pnpm@7.4.1 --activate" to update.     │
   │                                                                 │
   │     Follow @pnpmjs for updates: https://twitter.com/pnpmjs      │
   │                                                                 │
   ╰─────────────────────────────────────────────────────────────────╯

Lockfile is up-to-date, resolution step is skipped
Progress: resolved 1, reused 0, downloaded 0, added 0
Packages: +43
+++++++++++++++++++++++++++++++++++++++++++
Packages are cloned from the content-addressable store to the virtual store.
  Content-addressable store is at: /root/.local/share/pnpm/store/v3
  Virtual store is at:             node_modules/.pnpm
Progress: resolved 43, reused 0, downloaded 28, added 28
.../node_modules/vue-demi postinstall$ node ./scripts/postinstall.js
.../node_modules/vue-demi postinstall: Done

dependencies:
+ @vueuse/core 8.7.5
+ @vueuse/head 0.7.6
+ nprogress 0.2.0
+ pinia 2.0.14
+ prism-theme-vars 0.2.3
+ vue 3.2.37
+ vue-demi 0.13.1
+ vue-i18n 9.1.10
+ vue-router 4.0.16

devDependencies: skipped

Progress: resolved 43, reused 0, downloaded 43, added 43, done
--> f6588a70220
STEP 6/7: RUN pnpm build

> @ build /root/app
> vite-ssg build

sh: vite-ssg: not found
 ELIFECYCLE  Command failed.
error building at STEP "RUN pnpm build": error while running runtime: exit status 1
```
Looks like we still need dev deps to build the app ; )

Reply:
[npm dependencies and devDependencies](https://nodejs.dev/learn/npm-dependencies-and-devdependencies)
I assumed that dev deps are only used for development phase and vitesse package json respected that information which clearly didn't, I have no Idea why it ran perfectly fine when I tested the dockerfile 3 days ago. on a fresh clone it doesn't work as @hannoeru shows it. I would recommend moving every critical build dependency from dev deps to reqular deps in package.json.

Reply:
@Mhdi-kr What is the purpose of this Dockerfile? Production or Development?

Reply:
This is for production, I personally run my development tasks on my own machine rather than containers. @hannoeru

Reply:
https://github.com/antfu/vitesse/pull/380

Reply:
> I assumed that dev deps are only used for development phase and vitesse package json respected that information which clearly didn't

because `vite preview` is not made for production usage, we should build vite app first then serve it as static contents, so we don't need any dependencies on production.

>  I have no Idea why it ran perfectly fine when I tested the dockerfile 3 days ago. on a fresh clone.

I think is because you didn't exclude `node_modules` from docker build, so if you installed dependencies on you local machine then build the docker container the exist dependencies will still stay there, that is why the build didn't failed.